IQDropDownTextField/IQDropDownTextField.m

```diff
@@ -288,14 +288,8 @@ - (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComp
 
 - (NSInteger)selectedRow
 {
-    if (self.isOptionalDropDown)
-    {
-        return [self.pickerView selectedRowInComponent:0]-1;
-    }
-    else
-    {
-        return [self.pickerView selectedRowInComponent:0];
-    }
+    
+    return [self.pickerView selectedRowInComponent:0];
```

Comment:
If it's an optional dropdown then it will return wrong result in this case. Let's take an example of optional dropdown and user put 5 options, then if user select item 3 (index 2) then library should return index 2. But with optional dropdown we add another option at the top (which is actually an optional, so If I adopt your solution then selecting item 3 will return index 3 which isn't correct.